pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -115,164 +148,136 @@ decl_error! {
 
-					Ok(())
-				});
+				return Ok(().into());
 			}
 
 			Err(Error::<T>::UnsupportedCurrency.into())
 		}
 
-		/// Add additional currency to the list of supported currencies which fees can be paid in
-		/// Only selected members can perform this action
-		#[weight = (<T as Config>::WeightInfo::add_currency(), Pays::No)]
-		pub fn add_currency(origin, currency: AssetId) -> DispatchResult{
+		#[pallet::weight((<T as Config>::WeightInfo::add_currency(), DispatchClass::Normal, Pays::No))]
+		pub fn add_currency(origin: OriginFor<T>, currency: AssetId) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 
-			ensure!(
-				currency != CORE_ASSET_ID,
-				Error::<T>::CoreAssetNotAllowed
-			);
+			ensure!(currency != CORE_ASSET_ID, Error::<T>::CoreAssetNotAllowed);
 
 			// Only selected accounts can perform this action
-			ensure!(
-				Self::authorities().contains(&who),
-				Error::<T>::NotAllowed
-			);
-
-			if AcceptedCurrencies::mutate(|x| x.insert(currency)) {
-				Self::deposit_event(RawEvent::CurrencyAdded(who, currency));
-				return Ok(());
+			ensure!(Self::authorities().contains(&who), Error::<T>::NotAllowed);
+
+			if AcceptedCurrencies::<T>::mutate(|x| x.insert(currency)) {
+				Self::deposit_event(Event::CurrencyAdded(who, currency));
+				return Ok(().into());
 			}
 			Err(Error::<T>::AlreadyAccepted.into())
 		}
 
 		/// Remove currency from the list of supported currencies
 		/// Only selected members can perform this action
-		#[weight = (<T as Config>::WeightInfo::remove_currency(), Pays::No)]
-		pub fn remove_currency(origin, currency: AssetId) -> DispatchResult{
-
+		#[pallet::weight((<T as Config>::WeightInfo::remove_currency(), DispatchClass::Normal, Pays::No))]
+		pub fn remove_currency(origin: OriginFor<T>, currency: AssetId) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 
-			ensure!(
-				currency != CORE_ASSET_ID,
-				Error::<T>::CoreAssetNotAllowed
-			);
+			ensure!(currency != CORE_ASSET_ID, Error::<T>::CoreAssetNotAllowed);
 
 			// Only selected accounts can perform this action
-			ensure!(
-				Self::authorities().contains(&who),
-				Error::<T>::NotAllowed
-			);
-
-			if AcceptedCurrencies::mutate(|x| x.remove(&currency)) {
-				Self::deposit_event(RawEvent::CurrencyRemoved(who, currency));
-				return Ok(());
+			ensure!(Self::authorities().contains(&who), Error::<T>::NotAllowed);
+
+			if AcceptedCurrencies::<T>::mutate(|x| x.remove(&currency)) {
+				Self::deposit_event(Event::CurrencyRemoved(who, currency));
+				return Ok(().into());
 			}
 
 			Err(Error::<T>::UnsupportedCurrency.into())
 		}
 
 		/// Add an account as member to list of authorities who can manage list of accepted currencies
-		#[weight = (<T as Config>::WeightInfo::add_member(), Pays::No)]
-		pub fn add_member(origin, member: T::AccountId) -> DispatchResult{
+		#[pallet::weight((<T as Config>::WeightInfo::add_member(), DispatchClass::Normal, Pays::No))]
```

Comment:
Redundant ()?

Reply:
No. It is correct. The pallet::weight takes one argument - the same as before.

Reply:
Ah.. the change from weight = (a,b) to weight((a,b)) got me there.